src/controllers/species.controller.ts

```diff
@@ -58,12 +58,32 @@ export class SpeciesController {
   })
   async find(
     @param.query.object('filter', getFilterSchemaFor(Species))
-    filter?: Filter<Species>,
-  ): Promise<Species[]> {
+    filter?: Filter<Species> & { fields?: { captureCount?: Boolean } },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ): Promise<any[]> {
     console.log('get /species filter --> ', filter ? filter.where : null);
     // Only include active species
     filter = { ...filter, where: { ...filter?.where, active: true } };
-    return await this.speciesRepository.find(filter);
+
+    const captureCountStmt =
+      `SELECT species_id,COUNT(species_id) as count FROM trees ` +
+      `WHERE species_id NOTNULL AND active=true GROUP BY species_id`;
```

Comment:
How long time this sql would take?

Reply:
Roughly

Reply:
In the dev DB it's roughly 200ms in total